src/anim/evaluator/anim-evaluator.js

```diff
@@ -1,3 +1,130 @@
+import { Quat } from '../../math/quat.js';
+import { Vec3 } from '../../math/vec3.js';
+import { ANIM_LAYER_OVERWRITE } from '../controller/constants.js';
+
+class AnimTarget {
+    constructor(component, type) {
+        this._component = component;
+        this._mask = new Int8Array(component.layers.length);
+        this._weights = new Float32Array(component.layers.length);
+        this._totalWeight = 0;
+        this._counter = 0;
+        this._layerCounter = 0;
+        this._valueType = type;
+
+        if (this._valueType === AnimTarget.TYPE_QUAT) {
+            this._value = new Quat();
+            this._currentValue = new Quat();
+        } else {
+            this._value = new Vec3();
+            this._currentValue = new Vec3();
+        }
+    }
+
+    get valueType() {
+        return this._valueType;
+    }
+
+    get mask() {
+        return this._mask;
+    }
+
+    set mask(value) {
+        this._mask = value;
+    }
+
+    get weights() {
+        return this._weights;
+    }
+
+    get totalWeight() {
+        return this._totalWeight;
+    }
+
+    set totalWeight(value) {
+        this._totalWeight = value;
+    }
+
+    get counter() {
+        return this._counter;
+    }
+
+    set counter(value) {
+        this._counter = value;
+    }
+
+    get layerCounter() {
+        return this._layerCounter;
+    }
+
+    set layerCounter(value) {
+        this._layerCounter = value;
+    }
+
+    get value() {
+        return this._value;
+    }
+
+    set value(value) {
+        this._value = value;
+    }
```

Comment:
there's no need for these setters and getters, as they don't do anything extra to just having a property with the same name.